scripts/pr_previews/builder.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from __future__ import annotations
+
+import logging
+import shutil
+import subprocess
+from argparse import ArgumentParser
+from contextlib import contextmanager
+from typing import Iterator
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+
+IMAGE_NAME = "iqp-channel-docs-preview-builder"
+
+logger = logging.getLogger()
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    datefmt="%H:%M:%S",
+)
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+    parser.add_argument("dest", help="The output folder", type=Path)
+    return parser
+
+
+def main() -> None:
+    args = create_parser().parse_args()
+    with setup_dir() as dir:
+        yarn_build(dir)
+        save_output(dir, args.dest)
+
+
+def yarn_build(root_dir: Path) -> None:
+    # This ensures that dependencies like Sharp are properly installed. Most
+    # dependencies, like the first-party deps, will already have been installed.
+    _run_subprocess(["yarn", "install"], cwd=root_dir, stream_output=True)
+    _run_subprocess(["yarn", "build"], cwd=root_dir, stream_output=True)
+
+
+def save_output(root_dir: Path, dest: Path) -> None:
+    dest.mkdir(parents=True, exist_ok=True)
+    for item in (root_dir / "packages/preview/out").iterdir():
+        if item.is_dir():
+            shutil.copytree(item, dest / item.name)
+        else:
+            shutil.copy2(item, dest)
```

Comment:
Can this just be `shutil.copytree(root_dir / "packages/preview/out", dest)`?

Reply:
No because that would preserve the `/out` folder name. We want the children of `/out`, but not `out` itself.